x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,390 @@
+      if (dataViewIdRef && serializedState && !serializedState.dataViewId) {
+        serializedState.dataViewId = dataViewIdRef?.id;
+      }
+      return serializedState;
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const [coreStart, pluginStart] = await getStartServices();
+
+      const { http, uiSettings, notifications, ...startServices } = coreStart;
+      const { lens, data, usageCollection, fieldFormats } = pluginStart;
+
+      const deps = {
+        http,
+        uiSettings,
+        data,
+        notifications,
+        lens,
+        usageCollection,
+        fieldFormats,
+        ...startServices,
+      };
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        fieldStatsControlsApi,
+        dataLoadingApi,
+        fieldStatsControlsComparators,
+        serializeFieldStatsChartState,
+        onFieldStatsTableDestroy,
+        resetData$,
+      } = initializeFieldStatsControls(state);
+      const { onError, dataLoading, blockingError } = dataLoadingApi;
+
+      const defaultDataViewId = await deps.data.dataViews.getDefaultId();
+      const validDataViewId: string =
+        isDefined(state.dataViewId) && state.dataViewId !== ''
+          ? state.dataViewId
+          : defaultDataViewId ?? '';
+      let initialDataView: DataView[] | undefined;
+      try {
+        const dataView = isESQLQuery(state.query)
+          ? await getESQLAdHocDataview(
+              getIndexPatternFromESQLQuery(state.query.esql),
+              deps.data.dataViews
+            )
+          : await deps.data.dataViews.get(validDataViewId);
+        initialDataView = [dataView];
+      } catch (error) {
+        // Only need to publish blocking error if viewtype is data view, and no data view found
+        if (state.viewType === FieldStatsInitializerViewType.DATA_VIEW) {
+          onError(error);
+        }
+      }
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(initialDataView);
+
+      const subscriptions = new Subscription();
+      if (fieldStatsControlsApi.dataViewId$) {
+        subscriptions.add(
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => {
+                try {
+                  return deps.data.dataViews.get(dataViewId ?? defaultDataViewId);
+                } catch (error) {
+                  return of(undefined);
+                }
+              })
+            )
+            .subscribe((nextSelectedDataView) => {
+              if (nextSelectedDataView) {
```

Comment:
Fetching `initialDataView` sets a blocking error when data view id can not be found. Here, when data view id changes and data view can not be found, the error is ignored and no blocking error is set. Is this on purpose? Should switchMap return an error in the catch? Something more like

```
if (fieldStatsControlsApi.dataViewId$) {
  subscriptions.add(
    fieldStatsControlsApi.dataViewId$
      .pipe(
        skip(1),
        skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
        switchMap(async (dataViewId) => {
          try {
            const dataView = await deps.data.dataViews.get(dataViewId ?? defaultDataViewId);
            return { dataView };
          } catch (error) {
            return { dataView: undefined, error };
          }
        })
      )
      .subscribe(({ dataView, error }) => {
        if (error) {
          onError(error);
        }
        dataViews$.next([dataView]);
      })
  );
}
```

Reply:
I don't think this should be a blocking error. The table itself doesn't need any data view to render, we are fetching data views only for dashboard filters to show the values properly.

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/1cad8784ad2ecc05fa10040827a7a2fc0d434c7b

Reply:
> I don't think this should be a blocking error. The table itself doesn't need any data view to render, we are fetching data views only for dashboard filters to show the values properly.

Thanks for explaining.

I think the only change needed is that switchMap function needs to be async, `switchMap(async (dataViewId) => {}`. And then await dataViews.get for the return `return await deps.data.dataViews.get(dataViewId ?? defaultDataViewId);`

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/3fa980b4e50624d9b73d3b77c8249fccd1d07fd5